packages/app/src/hooks/useChartConfig.tsx

```diff
@@ -32,19 +79,66 @@ export function useQueriedChartConfig(
         console.log('mtViewDDL:', mtViewDDL);
         query = await renderMTViewConfig();
       }
-      if (query == null) {
-        query = await renderChartConfig(config, getMetadata());
+
+      const queries: ChSql[] = await Promise.all(
+        splitChartConfigs(config).map(c => renderChartConfig(c, getMetadata())),
+      );
+
+      const resultSets = await Promise.all(
+        queries.map(async query => {
+          const resp = await clickhouseClient.query<'JSON'>({
+            query: query.sql,
+            query_params: query.params,
+            format: 'JSON',
+            abort_signal: signal,
+            connectionId: config.connection,
+          });
+          return resp.json<any>();
+        }),
+      );
+
+      if (resultSets.length === 1) {
+        return resultSets[0];
       }
+      // join resultSets
+      else if (resultSets.length > 1) {
+        const metaSet = new Map<string, { name: string; type: string }>();
+        const tsBucketMap = new Map<string, Record<string, string | number>>();
+        for (const resultSet of resultSets) {
+          // set up the meta data
+          if (Array.isArray(resultSet.meta)) {
+            for (const meta of resultSet.meta) {
+              const key = meta.name;
+              if (!metaSet.has(key)) {
+                metaSet.set(key, meta);
+              }
+            }
+          }
 
-      const resultSet = await clickhouseClient.query<'JSON'>({
-        query: query.sql,
-        query_params: query.params,
-        format: 'JSON',
-        abort_signal: signal,
-        connectionId: config.connection,
-      });
+          const timestampColumn = inferTimestampColumn(resultSet.meta ?? []);
+          if (timestampColumn == null) {
+            throw new Error('No timestamp column');
```

Comment:
Are we also care the case when there is only one resultSet and `resultSet.meta` is missing `timestampColumn`?

Reply:
- single resultSet is handled here https://github.com/hyperdxio/hyperdx/blob/53f150685bf1100a18e154bc369a9b5448d9523e/packages/app/src/hooks/useChartConfig.tsx#L100-L102
- good catch on undefined `timestampColumn` which happens to table chart. pushed the fix